### PR TITLE
feat(sfs): add a new resource to manage SFS Turbo data task

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new resource to manage SFS Turbo data import or export task.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new resource to manage SFS Turbo data import or export task.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccDataTask_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccDataTask_basic -timeout 360m -parallel 4
=== RUN   TestAccDataTask_basic
=== PAUSE TestAccDataTask_basic
=== CONT  TestAccDataTask_basic
--- PASS: TestAccDataTask_basic (671.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       671.689s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image

     If the resource depends the parent resource(s), please provide the related check result(s) of the CheckDeleted validation.
    ab. Related resources (parent resources) not found
    \>>>>>> Paste the screenshot here <<<<<<
image

  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/resources/sfs_turbo_data_task.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  Changing this creates a new resource.
+  Currently, the value only support keep the same as `src_target`.
+
+* `src_prefix` - (Optional, String, ForceNew) Specifies source path prefix of the data task.
+  Changing this creates a new resource.
+  + For an import task, do not include the OBS bucket name.
+  + For an export task, do not include the name of the linkage directory.
+  + For a preload task, the source path prefix must be a directory or specific object ended with a slash (/).
+  + If not specified, all objects in the bound OBS bucket will be imported when importing, all files in the linkage
+  directory will be exported when exporting.
```
Comment (deer-hang) [inline docs/resources/sfs_turbo_data_task.md]:
```suggestion
    directory will be exported when exporting.
```

File: huaweicloud/services/acceptance/sfs/resource_huaweicloud_sfs_turbo_data_task_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+}
+`, testAccOBSTarget_basic(name, randInt))
+}
+
+func testAccDataTaskImportStateFunc(rName string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		var shareId, taskId string
+		rs, ok := s.RootModule().Resources[rName]
+		if !ok {
+			return "", fmt.Errorf("resource (%s) not found: %s", rName, rs)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/sfs/resource_huaweicloud_sfs_turbo_data_task_test.go]:
```suggestion
			return "", fmt.Errorf("resource (%s) not found", rName)
```
Field `rs` is nill when `ok` is false, it is unnecessary to print it.

File: huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_data_task.go [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+		Refresh:      dataTaskStatusRefreshFunc(d, meta, false),
+		Timeout:      t,
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	return err
+}
+
+func dataTaskStatusRefreshFunc(d *schema.ResourceData, meta interface{}, isDelete bool) resource.StateRefreshFunc {
```
Comment (deer-hang) [inline huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_data_task.go]:
```suggestion
func dataTaskStatusRefreshFunc(d *schema.ResourceData, meta interface{}) resource.StateRefreshFunc {
```

File: huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_data_task.go [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+	return func() (interface{}, string, error) {
+		resp, err := getDataTaskInfo(d, meta)
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok && isDelete {
```
Comment (deer-hang) [inline huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_data_task.go]: Please check this 404 error.

File: huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_data_task.go [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+		}
+
+		respBody, err := utils.FlattenResponse(resp)
+		if err != nil {
+			return nil, "ERROR", err
+		}
+
+		state, err := jmespath.Search("status", respBody)
+		if err != nil {
+			return nil, "ERROR", fmt.Errorf("error parsing %s from response body", state)
```
Comment (deer-hang) [inline huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_data_task.go]: why printing state?

File: huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_data_task.go [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+		}
+
+		statusRaw := fmt.Sprintf("%v", state)
+		if utils.StrSliceContains([]string{"SUCCESS", "FAIL"}, statusRaw) {
```
Comment (deer-hang) [inline huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_data_task.go:245]: add a note for COMPLETED status.

File: docs/resources/sfs_turbo_data_task.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+page_title: "HuaweiCloud: huaweicloud_sfs_turbo_data_task"
+description: |-
+  Manages a data import or export task resource under the SFS Turbo within HuaweiCloud.
+---
+
+# huaweicloud_sfs_turbo_data_task
+
+Manages a data import or export task resource under the SFS Turbo within HuaweiCloud.
+
+-> Destroying resources does not change the current state.
```
Comment (deer-hang) [inline docs/resources/sfs_turbo_data_task.md]:
```suggestion
-> Destroying resources does not change the current state. Please pay attention to the results of task execution through `status`.
```

File: huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_data_task.go [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+	d.SetId(parts[1])
+	mErr := multierror.Append(nil,
+		d.Set("share_id", parts[0]),
+	)
+	return []*schema.ResourceData{d}, mErr.ErrorOrNil()
+}
+
+// When the data task does not exist, the response body example of the details interface is as follows:
+// error message: {"errCode":"SFS.TURBO.0106","errMsg":"Invalid task id."}
+func hasSpecifyError(err error, specCode string) bool {
```
Comment (deer-hang) [inline huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_data_task.go]: Please use public methods first.
The second parameter is unnecessary

File: docs/resources/sfs_turbo_data_task.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+  If omitted, the provider-level region will be used. Changing this creates a new resource.
+
+* `share_id` - (Required, String, ForceNew) Specifies the ID of the SFS Turbo file system to which the data task
+  belongs. Changing this creates a new resource.
+
+* `type` - (Required, String, ForceNew) Specifies the type of the data task.
+  Changing this creates a new resource.
+  The valid values are as following:
+  + **import**: Additional metadata import. This import method will import the metadata (name, size, and latest update
+  time) of the OBS object and the additional metadata (such as uid, gid and mode) exported from SFS Turbo.
```
Comment (jinyangyang222) [inline docs/resources/sfs_turbo_data_task.md:47]:
```suggestion
    time) of the OBS object and the additional metadata (such as uid, gid and mode) exported from SFS Turbo.
```

Comment (jinyangyang222) [reply docs/resources/sfs_turbo_data_task.md:47]: Please also check the same format elsewhere

File: huaweicloud/services/acceptance/sfs/resource_huaweicloud_sfs_turbo_data_task_test.go [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccDataTaskImportStateFunc(rName),
+			},
+		},
+	})
+}
+
+func testAccDataTask_basic(name string, randInt int) string {
```
Comment (jinyangyang222) [inline huaweicloud/services/acceptance/sfs/resource_huaweicloud_sfs_turbo_data_task_test.go:86]: Test cases also need to cover other optional parameters

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccDataTask_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccDataTask_basic -timeout 360m -parallel 4
=== RUN   TestAccDataTask_basic
=== PAUSE TestAccDataTask_basic
=== CONT  TestAccDataTask_basic
--- PASS: TestAccDataTask_basic (672.97s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       673.064s
```

**Pull request merged**